<launch>

    <node pkg="tf_tree_publisher" type="tf_tree_publisher_node" name="tf_tree_publisher"  output="screen" >
        
        <param name="num_of_ins"            	type="int"  	value="1" />
        <param name="num_of_lidar"          	type="int"      value="2" />
        <param name="num_of_camera"         	type="int"      value="1" />
        <param name="num_of_radar"          	type="int"      value="0" />

        <param name="world_frame"           	type="str"      value="world" />
        <param name="map_frame"             	type="str"      value="map" />
        <param name="ins_frame"             	type="str"      value="ins" />
        <param name="base_link_frame"       	type="str"      value="baselink" />
        <param name="lidar_frame"           	type="str"      value="os_sensor" />
        <param name="roof_lidar_frame"          type="str"      value="roof/os_sensor" /> 
        <param name="camera_frame"          	type="str"      value="camera" />
        <param name="radar_frame"           	type="str"      value="radar" />
        <param name="friction_frame"           	type="str"      value="friction_image" />

        <param name="map_translation_x"     	type="double"   value="741331.804017" />
        <param name="map_translation_y"     	type="double"   value="3391511.58541" />
        <param name="map_translation_z"     	type="double"   value="49.0" />
        <param name="map_rotation_roll"     	type="double"   value="0.0" />
        <param name="map_rotation_pitch"    	type="double"   value="0.0" />
        <param name="map_rotation_yaw"      	type="double"   value="0.0" />

        <param name="base_link_translation_x"	type="double"   value="-0.4699" />
        <param name="base_link_translation_y"	type="double"   value="0.0" />
        <param name="base_link_translation_z" 	type="double"   value="0.0" />
        <param name="base_link_rotation_roll" 	type="double"   value="0.0" />
        <param name="base_link_rotation_pitch"	type="double"   value="0.0" />
        <param name="base_link_rotation_yaw"    type="double"   value="0.0" />

        <param name="lidar_translation_x"     	type="double"   value="3.0226" />
        <param name="lidar_translation_y"     	type="double"   value="0.3556" />
        <param name="lidar_translation_z"     	type="double"   value="-0.3556" />
        <param name="lidar_rotation_roll"     	type="double"   value="0.0" />
        <param name="lidar_rotation_pitch"    	type="double"   value="0.0" />
        <param name="lidar_rotation_yaw"      	type="double"   value="0.0" />

        <param name="roof_lidar_translation_x"       type="double"   value="0.8" />
        <param name="roof_lidar_translation_y"       type="double"   value="0.0" />
        <param name="roof_lidar_translation_z"       type="double"   value="0.8" />
        <param name="roof_lidar_rotation_roll"       type="double"   value="0.0" />
        <param name="roof_lidar_rotation_pitch"      type="double"   value="0.0" />
        <param name="roof_lidar_rotation_yaw"        type="double"   value="0.0" />

        <param name="camera_translation_x"     	type="double"   value="0.8001" />
        <param name="camera_translation_y"     	type="double"   value="0.0" />
        <param name="camera_translation_z"     	type="double"   value="0.9017" />
        <param name="camera_rotation_roll"     	type="double"   value="0.0" />
        <param name="camera_rotation_pitch"    	type="double"   value="0.0" />
        <param name="camera_rotation_yaw"      	type="double"   value="0.0" />

        <param name="friction_translation_x"     	type="double"   value="5.79543333164" />
        <param name="friction_translation_y"     	type="double"   value="0.0" />
        <param name="friction_translation_z"     	type="double"   value="0.0" />
        <param name="friction_rotation_roll"     	type="double"   value="0.0" />
        <param name="friction_rotation_pitch"    	type="double"   value="0.0" />
        <param name="friction_rotation_yaw"      	type="double"   value="0.0" />
        
        <remap from="/odometry_data" to="gv01/odom" />

    </node>

</launch>
